### ISSUE TITLE
flaky test: iroh-net endpoint::tests::save_load_peers

### ISSUE THREAD

**Issue by dignifiedquire** [n0-computer/iroh#2504]
```
test endpoint::tests::save_load_peers ... FAILED

failures:

failures:
    endpoint::tests::save_load_peers

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 106 filtered out; finished in 11.83s


--- STDERR:              iroh-net endpoint::tests::save_load_peers ---
thread 'endpoint::tests::save_load_peers' panicked at iroh-net\src\endpoint.rs:1302:82:
called `Option::unwrap()` on a `None` value
```

https://github.com/n0-computer/iroh/actions/runs/9939026148/job/27452760889

**Comment by flub** [comment]
The flakyness here is a race condition between the NodeMap loading the peers from the file and the test requesting this info.  It will likely be solved by #2510 and is probably best to address there.

**Comment by dignifiedquire** [comment]
Test is gone
